tests/test_crud_data.py

```diff
@@ -0,0 +1,348 @@
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False)
+
+
+class DataCRUDDataAccess(CRUDDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('name', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('active', bool),
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, Data, connect_to_mem_db(), DataCRUDDataAccess.rules_validator)
+
+
+class DataSoftDelete(Base, SoftDeleteMixin):
+    __tablename__ = 'crud_softdel_test'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False)
+
+
+class DataCRUDSoftDeleteDataAccess(CRUDSoftDeleteDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('name', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('active', bool),
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, DataSoftDelete, connect_to_mem_db(), DataCRUDSoftDeleteDataAccess.rules_validator)
+
+
+class DataSoftDeleteToken(Base, SoftDeleteMixin, SoftDeleteTokenMixin):
+    __tablename__ = 'crud_softdel_token_test'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False)
+
+
+class DataCRUDSoftDeleteTokenDataAccess(CRUDSoftDeleteWithTokenDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('name', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('active', bool),
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, DataSoftDeleteToken, connect_to_mem_db(), DataCRUDSoftDeleteTokenDataAccess.rules_validator)
+
+
+class TestCrud(unittest.TestCase):
+    def test_crud(self):
+        crud = DataCRUDDataAccess()
+
+        crud.create({'name': 'test_name', 'email': 'abc@def.com', 'active': True})
+
+        data = result_to_dict(crud.get(1))
+        self.assertNotEqual(data, None)
+        self.assertDictEqual(data, {'id': 1, 'name': 'test_name', 'email': 'abc@def.com', 'active': True})
+
+        crud.update(1, {'active': False})
+
+        data = result_to_dict(crud.get(1))
+        self.assertNotEqual(data, None)
+        self.assertDictEqual(data, {'id': 1, 'name': 'test_name', 'email': 'abc@def.com', 'active': False})
+
+        crud.delete(1)
+
+        with self.assertRaises(StatusCodeException):
+            crud.get(1)
+
+    def test_crud_multiple(self):

```

Comment:
thank you looks good. 

what do you think about. was just thinking

test_crud_multiple, test_crud_soft_delete, test_crud_soft_delete_token
are duplication of code but to the deleted_at. updated_at, delete_token. 

i was thinking can we create a private test method for example 
_test_data_access(data_access: DataAccess, is_soft_delete: bool, is_token_delete: bool):
   ... test single + test multi